src/amo/components/AddonQRCode/index.js

```diff
@@ -0,0 +1,184 @@
+import qrCode328839 from './img/328839.png';
+import qrCode506646 from './img/506646.png';
+import qrCode520576 from './img/520576.png';
+import qrCode521554 from './img/521554.png';
+import qrCode607454 from './img/607454.png';
+import qrCode627490 from './img/627490.png';
+import qrCode722 from './img/722.png';
+import qrCode735894 from './img/735894.png';
+import qrCode811592 from './img/811592.png';
+import qrCode824288 from './img/824288.png';
+import qrCode855413 from './img/855413.png';
+import qrCode866226 from './img/866226.png';
+import qrCode869140 from './img/869140.png';
+import qrCode953945 from './img/953945.png';
+import qrCode9609 from './img/9609.png';
+
+export const qrCodeSrcs = {
+  '229918': qrCode229918,
+  '328839': qrCode328839,
+  '506646': qrCode506646,
+  '520576': qrCode520576,
+  '521554': qrCode521554,
+  '607454': qrCode607454,
+  '627490': qrCode627490,
+  '722': qrCode722,
+  '735894': qrCode735894,
+  '811592': qrCode811592,
+  '824288': qrCode824288,
+  '855413': qrCode855413,
+  '866226': qrCode866226,
+  '869140': qrCode869140,
+  '953945': qrCode953945,
+  '9609': qrCode9609,
+};
+
+export const ADDON_QRCODE_CAMPAIGN = 'addon-qr-code';
+export const ADDON_QRCODE_CATEGORY = 'Addon QR Code';
+export const ADDON_QRCODE_CLICK_ACTION = 'addon-qr-code-click';
+export const ADDON_QRCODE_IMPRESSION_ACTION = 'addon-qr-code-impression';
+
+type Props = {|
+  addon: AddonType,
+  onDismiss: (e: ElementEvent | null) => void,
+|};
+
+export type DeafultProps = {|
+  _tracking: typeof tracking,
+|};
+
+export type InternalProps = {|
+  ...Props,
+  ...DeafultProps,
+  i18n: I18nType,
+|};
+
+export class AddonQRCodeBase extends React.Component<InternalProps> {
+  static defaultProps: DeafultProps = {
+    _tracking: tracking,
+  };
+
+  onInteract: (action: string) => void = (action) => {
+    const { _tracking, addon } = this.props;
+
+    _tracking.sendEvent({
+      action,
+      category: ADDON_QRCODE_CATEGORY,
+      label: String(addon.id),
+    });
+  };
+
+  onLinkClick: () => void = () => {
+    this.onInteract(ADDON_QRCODE_CLICK_ACTION);
+    this.props.onDismiss(null);
+  };
+
+  componentDidMount() {
+    this.onInteract(ADDON_QRCODE_IMPRESSION_ACTION);
+  }
+
+  render(): React.Node {
+    const { addon, i18n, onDismiss } = this.props;
+
+    const idAsString = String(addon.id);
+    const imgSrc = qrCodeSrcs[idAsString];
+    invariant(imgSrc, `Could not find a QR code for addonId: ${idAsString}`);
+
+    let addonUrl = addQueryParams(getAddonURL(addon.slug), {
+      utm_source: DEFAULT_UTM_SOURCE,
+      utm_medium: DEFAULT_UTM_MEDIUM,
+      utm_campaign: ADDON_QRCODE_CAMPAIGN,
+      utm_content: String(idAsString),
+    });
+    addonUrl = `/${CLIENT_APP_ANDROID}${addonUrl}`;
+
+    const labelText = i18n.sprintf(
+      i18n.gettext(
+        'To get %(addonName)s on Firefox for Android, point your device camera to the code above or copy %(linkStart)sthis link%(linkEnd)s',
```

Comment:
I'm aware this is from the PRD but I'm not sure this is descriptive/accurate enough - it reads to me like you should open the camera app on your phone, which won't scan the QR code - you need to choose the scan button within Firefox.

Reply:
If you have an iPhone, simply opening the camera and pointing it at a QR code *will* scan it, at least on my phone. I imagine that's the scenario that we are looking at here. I suppose to be more specific we could say "scan the code above with your camera" or something like that? I'm not sure how we came up with the instructions for that.

Reply:
if you have an IPhone, can you install addons?

Reply:
> if you have an iPhone, can you install addons?

Um, good point! I don't have an Android phone so I guess I'm not the target audience. 😉 What is the experience like on an Android phone, I guess that's what we need to know.

Reply:
> I'm not sure how we came up with the instructions for that.

If I had to guess, I'd say the person who came up with instructions had an iPhone and assumed the experience was the same on other platforms 😬

The experience for me on Android (specifically for me - from a quick search the behaviour across different manufactures and models is [gloriously inconsistent](https://www.qr-code-generator.com/guides/scan-qr-code-android/)) is the camera app doesn't automatically detect the QR code - I either have to press a button in the camera app to instead open a Lens app that has all kinds of image detection built in to it, or use the QR button in Firefox itself (what I would do).  
I wonder if there's a support article already about QR support in Firefox we could link to

Reply:
> Um, good point! I don't have an Android phone 

-> service-now request 😉

Reply:
On my Android phone(s), I have to install a "QR Scanner" app, which is then used directly or by other apps to scan QR codes.

Reply:
> On my Android phone(s), I have to install a "QR Scanner" app, which is then used directly or by other apps to scan QR codes.

For you, Firefox doesn't have any support for scanning QR codes without manually installing a 3rd party app?

Reply:
ha, I dunno. I wasn't aware of this feature in Fenix before today.

Reply:
Because the experience can be so different on different devices, and with different apps, perhaps it should just say "scan the code above with your phone"?

Reply:
> Because the experience can be so different on different devices, and with different apps, perhaps it should just say "scan the code above with your phone"?

+1

Reply:
Meridel has agreed to take a look at this for us and provide an opinion on what copy we should use for this. I think we can land this as is for now, and I will open a follow-up to discuss changing the copy.

Reply:
I opened https://github.com/mozilla/addons-frontend/issues/10910 as the follow-up.